### DISCUSSION TITLE
Using ML as anticheat

### DISCUSSION THREAD

**Discussion by leg7** [ppy/osu#23630] [Ideas] [answered]
Given how much data osu! has of user plays and how easy it is to analyze I think it would make sense to consider one of the new ML anticheats. I'm not sure how anticheat is implemented in osu! and how people cheat but the game has had it's fair share of cheating scandals in the past, and osu! going FOSS certainly will make the situation easier for cheaters.

Solutions are already out there that could be run server side like https://anybrain.gg/ (absolutely proprietary) or FOSS stuff like https://waldo.vision/.
I found these projects through this [video](https://youtu.be/LkmIItTrQP4).

Of course implementing a custom solution for osu! would be non-trivial and require good ML knowledge but the benefits would be great. I also don't think it would be much harder to implement than a comprehensive and invasive local anticheat. The models would also have to be trained on 100% correct data but I digress.

**Comment by monochrome22** [comment, marked as answer]
Should be duplicate of https://github.com/ppy/osu/discussions/23194.
